prov/rxm/src/rxm_ep.c

```diff
@@ -1360,7 +1360,10 @@ rxm_ep_send_common(struct rxm_ep *rxm_ep, struct rxm_conn *rxm_conn,
 				rxm_ep_do_progress(&rxm_ep->util_ep);
 			rxm_tx_buf_release(rxm_ep, RXM_BUF_POOL_TX, tx_buf);
 		}
-	} else if (data_len <= rxm_ep->sar_limit) {
+	} else if (data_len <= rxm_ep->sar_limit &&
+		   /* SAR uses eager_limit as segment size */
+		   (rxm_ep->eager_limit <
+		    (1ULL << (8 * sizeof_field(struct ofi_ctrl_hdr, seg_size))))) {
```

Comment:
I think you want 2 ^ sizeof_field.  If the field is 8-bits, you have 2^8 = 256 bytes, not 8*8 = 64 bytes.

Reply:
`sizeof_field` returns the size in bytes so we multiply it by 8 to get the # of bits and then use it to left shift `1ULL` to get the max value that can be represented by the field + 1.